src/main/java/com/google/api/codegen/GapicContext.java

```diff
@@ -36,42 +36,49 @@
 public class GapicContext extends CodegenContext {
 
   private final Model model;
-  private final ApiConfig apiConfig;
+  private final GapicProductConfig productConfig;
 
   private final ServiceMessages serviceMessages;
-  private final ServiceConfig serviceConfig;
+  private final ProductServiceConfig productServiceConfig;
 
   public static final String API_WRAPPER_SUFFIX = "Client";
 
   /** Constructs the abstract instance. */
-  protected GapicContext(Model model, ApiConfig apiConfig) {
+  protected GapicContext(Model model, GapicProductConfig productConfig) {
     this.model = Preconditions.checkNotNull(model);
-    this.apiConfig = Preconditions.checkNotNull(apiConfig);
+    this.productConfig = Preconditions.checkNotNull(productConfig);
     this.serviceMessages = new ServiceMessages();
-    this.serviceConfig = new ServiceConfig();
+    this.productServiceConfig = new ProductServiceConfig();
   }
 
   /** Returns the associated model. */
   public Model getModel() {
     return model;
   }
 
-  /** Returns the associated config. */
-  public ApiConfig getApiConfig() {
-    return apiConfig;
+  /*
+   * Returns the associated config.
+   *
+   * NOTE: The name here is out of date, but this whole class will be deprecated soon.
+   */
+  public GapicProductConfig getApiConfig() {
+    return productConfig;
   }
 
   public ServiceMessages messages() {
     return serviceMessages;
   }
 
-  public ServiceConfig getServiceConfig() {
-    return serviceConfig;
+  /*
+   * NOTE: The name here is out of date, but this whole class will be deprecated soon.
+   */
+  public ProductServiceConfig getServiceConfig() {
+    return productServiceConfig;
   }
 
-  /** Return the name of the class which is the GAPIC wrapper for this service interface. */
-  public String getApiWrapperName(Interface service) {
-    return service.getSimpleName() + API_WRAPPER_SUFFIX;
+  /** Return the name of the class which is the GAPIC wrapper for this API interface. */
+  public String getApiWrapperName(Interface apiInterface) {
```

Comment:
The compound term "API interface" has a clear referent, even if the components (API, interface) don't. 

We can't rename `Interface`, it comes from api-compiler.